internal/status/clock.go

```diff
@@ -0,0 +1,37 @@
+package status
+
+import "time"
+
+// Clock returns the current local time.
+type Clock interface {
+	Now() time.Time
+}
+
+// Real clock returns the current local time.
+type RealClock struct {
+}
+
+// NewRealClock creates a new RealClock.
+func NewRealClock() *RealClock {
+	return &RealClock{}
+}
+
+// Now returns the current local time.
+func (c *RealClock) Now() time.Time {
+	return time.Now()
+}
+
+// FakeClock allows you to control the returned time.
+type FakeClock struct {
```

Comment:
Why not use counterfeiter here to generate the fake?

Reply:
I used the old code from https://github.com/nginxinc/nginx-kubernetes-gateway/blob/eca8cd46a5b96ae8678b11a9dffc2988b9e275af/internal/state/clock.go which was written before we start using fake generation

it makes sense to generate 👍 

Note that I changed prepareGatewayStatus and prepareHTTPRouteStatus to accept time now, not a clock. This is because if I export the fake clock in the tests for those functions (which are in the status package), it will create a dependency cycle (statusfakes export status package). But I think the new function signature are better even without the need to break the cycle.
I also updated the clock interface to return metav1.Time to avoid unnecessary converstions time.Time -> metav1.Time